src/targets/gpu/kernels/include/migraphx/kernels/scatternd.hpp

```diff
@@ -0,0 +1,81 @@
+#ifndef MIGRAPHX_GUARD_KERNELS_SCATTERND_HPP
+#define MIGRAPHX_GUARD_KERNELS_SCATTERND_HPP
+
+#include <migraphx/kernels/index.hpp>
+#include <migraphx/kernels/algorithm.hpp>
+
+namespace migraphx {
+
+struct assign_none
+{
+    template <class T, class U>
+    MIGRAPHX_DEVICE_CONSTEXPR void operator()(T& x, U y) const
+    {
+        x = y;
+    }
+};
+
+struct assign_add
+{
+    template <class T, class U>
+    MIGRAPHX_DEVICE_CONSTEXPR void operator()(T& x, U y) const
+    {
+        x += y;
+    }
+};
+
+struct assign_mul
+{
+    template <class T, class U>
+    MIGRAPHX_DEVICE_CONSTEXPR void operator()(T& x, U y) const
+    {
+        x *= y;
+    }
+};
+
+template <class T, class U, class V, class W, class F>
+__device__ void
+scatternd(const T& /* data_t */, const U& indices_t, const V& updates_t, const W& output_t, F f)
+{
+    auto index         = make_index();
+    auto i             = index.global;
+    auto updates_shape = updates_t.get_shape();
+
+    if(i < updates_shape.elements())
```

Comment:
This probably needs to use a global strided loop. Otherwise this would miss elements if `nglobal < updates_shape.elements()`.